### PR TITLE
Update ci, rust version and grpcio version 

### PR DESCRIPTION
**Pull request by devillove084**

### CODE REVIEW

File: etcd-client/src/lib.rs [hunk 1]
```diff
@@ -82,7 +78,7 @@
     clippy::module_name_repetitions, // repeation of module name in a struct name is not big deal
     clippy::multiple_crate_versions, // multi-version dependency crates is not able to fix
     clippy::panic, // allow debug_assert, panic in production code
-    clippy::unknown_clippy_lints,  // allow rustc and clippy verison mismatch
+    // clippy::unknown_clippy_lints,  // allow rustc and clippy verison mismatch
```
Comment (pzheng1025) [inline etcd-client/src/lib.rs]: delete this

File: etcd-client/src/watch/mod.rs [hunk 2]
```diff
@@ -189,6 +185,7 @@ impl Watch {
         Self { client, tunnel }
     }
 
+    /// # Panics
```
Comment (pzheng1025) [inline etcd-client/src/watch/mod.rs:190]: Please add scenarios in which panicking occurs.

File: mock-etcd/src/mock_etcd.rs [hunk 3]
```diff
@@ -67,6 +67,7 @@ pub struct MockEtcdServer {
 }
 
 impl MockEtcdServer {
+    /// # Panics
```
Comment (pzheng1025) [inline mock-etcd/src/mock_etcd.rs:72]: Please add scenarios in which panicking occurs.

File: etcd-client/src/lib.rs (hunk 1 above)
Comment (devillove084) [reply etcd-client/src/lib.rs]: Done.

File: etcd-client/src/watch/mod.rs (hunk 2 above)
Comment (devillove084) [reply etcd-client/src/watch/mod.rs:190]: Done.

File: mock-etcd/src/mock_etcd.rs (hunk 3 above)
Comment (devillove084) [reply mock-etcd/src/mock_etcd.rs:72]: Done.

File: etcd-client/src/kv/delete.rs [hunk 4]
```diff
@@ -52,10 +52,11 @@ impl EtcdDeleteResponse {
     /// Takes the header out of response, leaving a `None` in its place.
     #[inline]
     pub fn take_header(&mut self) -> Option<ResponseHeader> {
-        match self.proto.header.take() {
-            Some(header) => Some(From::from(header)),
-            None => None,
-        }
+        // match self.proto.header.take() {
```
Comment (pzheng1025) [inline etcd-client/src/kv/delete.rs]: delete it

File: etcd-client/src/kv/mod.rs [hunk 5]
```diff
@@ -407,9 +407,11 @@ impl EtcdKeyValue {
     /// Takes the key out of response, leaving an empty vector in its place.
     #[inline]
     pub fn take_key(&mut self) -> Vec<u8> {
-        std::mem::replace(&mut self.proto.key, vec![])
+        std::mem::take(&mut self.proto.key)
+        //std::mem::replace(&mut self.proto.key, vec![])
```
Comment (pzheng1025) [inline etcd-client/src/kv/mod.rs]: delete it

File: etcd-client/src/lazy.rs [hunk 6]
```diff
@@ -192,7 +188,7 @@ mod tests {
             assert_eq!(calls.load(ORDER), 0, "Expected thunk not called.");
 
             let lock = lazy.read().await;
-            assert_eq!((*lock), true, "Expected read() == thunk().");
+            assert!(*lock);
```
Comment (pzheng1025) [inline etcd-client/src/lazy.rs]: assert!(*lock, "Expected read() == thunk().");

File: etcd-client/src/kv/mod.rs [hunk 7]
```diff
@@ -407,11 +407,16 @@ impl EtcdKeyValue {
...
     }
-
+
```
Comment (pzheng1025) [inline etcd-client/src/kv/mod.rs]: delete space

File: etcd-client/src/kv/mod.rs [hunk 8]
```diff
@@ -557,12 +565,13 @@ impl KeyRange {
...
-        std::mem::replace(&mut self.key, vec![])
+        std::mem::take(&mut self.key)
     }
 
     /// Take `range_end` value
     #[inline]
     pub fn take_range_end(&mut self) -> Vec<u8> {
-        std::mem::replace(&mut self.range_end, vec![])
+        std::mem::take(&mut self.range_end)
+        //std::mem::replace(&mut self.range_end, vec![])
```
Comment (pzheng1025) [inline etcd-client/src/kv/mod.rs]: delete it

Comment (devillove084) [reply etcd-client/src/kv/mod.rs]: done

**Review by pzheng1025 — APPROVED** [review]

**Pull request merged**